snooty/postprocess.py

```diff
@@ -365,6 +366,13 @@ def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
                     )
                 return
 
+        # Check for duplicate inclusions in this root page context.
+        if collections.Counter(fileid_stack).most_common(1)[0][1] > 1:
```

Comment:
will this pick up multiple duplicate inclusions if we are only checking the first one? or would we want to include all of loops?

Reply:
Hm, do you have a concrete example of the case you're thinking of?

In the case where a page contains multiple include loops, this logic does already warn about each. We only take the `most_common(1)` because that tells us as soon as a loop starts, and prevents recursing deeper.

Reply:
ah you're right. i've tested with multiple loops within the same root file and it prints multiple warnings:

```
E             +  (<class 'snooty.diagnostics.IncludeLoop'>,
E             +   'Includes loop: index.txt -> foo.rst -> index.txt'),
E             +  (<class 'snooty.diagnostics.IncludeLoop'>,
E             +   'Includes loop: index.txt -> foo2.rst -> foo3.rst -> index.txt'),
E             +  (<class 'snooty.diagnostics.IncludeLoop'>,
E             +   'Includes loop: index.txt -> foo2.rst -> foo3.rst -> index.txt -> foo.rst -> '
E             +   'index.txt'),
E             +  (<class 'snooty.diagnostics.IncludeLoop'>,
E             +   'Includes loop: index.txt -> foo2.rst -> foo3.rst -> index.txt -> foo.rst -> '
E             +   'index.txt'),
E             +  (<class 'snooty.diagnostics.IncludeLoop'>,
E             +   'Includes loop: index.txt -> foo2.rst -> foo3.rst -> index.txt'),
E               ]
```

notably its still sending the full fileid_stack!

Reply:
perhaps we can include a multi loop? not a big deal, it was my oversight!